.github/workflows/gpuci.yml

```diff
@@ -0,0 +1,47 @@
+name: "GPU CI/CD"
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+      - '[rv][0-9].[0-9].[0-9]'
+      - '[rv][0-9].[0-9].[0-9]rc[0-9]'
+    types: [ labeled ]
+
+jobs:
+  build-container:
+    if: ${{ github.event.label.name == 'gpuci' }}
+    uses: ./.github/workflows/_build_container.yml
+
+  run-gpu-tests:
+    needs: build-container
+    runs-on: self-hosted
+    if: ${{ github.event.label.name == 'gpuci' }}
+
+    steps:
+    - name: Remove existing container if it exists
+      run: |
+        if [ "$(docker ps -aq -f name=nemo-curator-container)" ]; then
+            docker rm -f nemo-curator-container
+        fi
+
+    - name: Run Docker container
+      run: |
+        docker run --gpus all --runtime=nvidia --name nemo-curator-container -d nemoci.azurecr.io/nemo_curator_container:${{ github.run_id }} bash -c "sleep infinity"
+
+    - name: Check GPUs
+      run: |
+        whoami
+        docker exec nemo-curator-container nvidia-smi
+
+    - name: Verify installations
+      run: |
+        docker exec nemo-curator-container /bin/bash -c "source activate rapids && pip list"
```

Comment:
Probably update the environment name to Curator here as well if it's updated in the docker file.